### PR TITLE
Templates for Shibboleth and OpenLDAP - single VM and clustered deployments

### PR DESCRIPTION
**Pull request by vinhub**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vinhub__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: openldap-cluster-ubuntu/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,460 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
```
Comment (singhkay) [inline openldap-cluster-ubuntu/azuredeploy.json]: Consider removing this. See #944

File: openldap-cluster-ubuntu/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+      "name": "[concat(concat(variables('vmName'),copyIndex()),'/newuserscript')]",
+      "apiVersion": "[variables('apiVersion')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachines/',concat(variables('vmName'),copyIndex()))]"
+      ],
+      "properties": {
+        "publisher": "Microsoft.OSTCExtensions",
+        "type": "CustomScriptForLinux",
+        "typeHandlerVersion": "1.4",
```
Comment (singhkay) [inline openldap-cluster-ubuntu/azuredeploy.json]: Add a autoUpgradeMinorVersion property

File: openldap-cluster-ubuntu/config_1_loadSyncProvModule.ldif [hunk 3]
```diff
@@ -0,0 +1,4 @@
+dn: cn=module{0},cn=config
```
Comment (singhkay) [inline openldap-cluster-ubuntu/config_1_loadSyncProvModule.ldif]: Please create a folder for these files as there are so many of them. It's becoming hard to locate the template json files among these.

**Comment by singhkay** [comment]
@vinhub Left feedback on the first template. Same should apply to others as well.

**Comment by vinhub** [comment]
@singhkay We have fixed all the issues you identified. The remaining issue is that of the Travis CI failure above. This failure is occurring because we refer to the install script (and other files) in the azuredeploy.json using a base path like  https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/openldap-singlevm-ubuntu/. But at present there are no files there because the files are a part of this PR. Once the PR is merged in, the files will appear at that path and then the CI will succeed. 

The alternative for us would be to change the path temporarily to point to our current repo, have the PR get merged and then send another PR which will update just the path. Please let us know how you would like to proceed.

**Comment by sedouard** [comment]
no, we do relinking to your fork. so it should work. its probably because your links are:

`https://raw.githubusercontent.com/azure/azure-quickstart-templates/master/openldap-singlevm-ubuntu/`

and not
`https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/openldap-singlevm-ubuntu/`

(not the casing of `Azure`)

**Comment by azurecla** [comment]
Hi __@vinhub__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
Hi __@vinhub__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by singhkay** [comment]
@vinhub Can you commit a small change to trigger a CI build?

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**